posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,506 @@
+
+The target location may be a packaged application archive -- EAR, WAR, RAR, or JAR files -- or may be a directory containing the fully or partially expanded contents of one of these archives.  The target can also be a directory containing java source, or a directory containing several application archives.  All of the artifacts in the target location are processed by the transformer.
+
+=== More about transforming server configurations
+
+Transformation of a server configuration is only necessary when java EE package names are exposed in the server configuration.  One case of explosed java EE packages names are JMS activation specification elements.  For example:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="javax.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+This must be transformed to:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="jakarta.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+== Updating features in server configurations:
+
+When running a Jakarta EE 9 enable application with Open-Liberty, Jakarta EE 9 versions of Open-Liberty features must be specified in the server configuration.
+
+In many cases, Jakarta EE 9 features use the same short names as the corresponding Java EE features.  The Jakarta EE 9 features simply update the feature version.  Here is a complete table of (non-internal) features which were updated by changing just the feature version:
+
+.Java EE to Jakarta EE 9 Feature Updates: Version Only
+|===
+| Java EE 8 feature name | Jakarta EE 9 feature name
+
+| appClientSupport-1.0
+| appClientSupport-2.0
+
+| appSecurity-3.0
+| appSecurity-4.0
+
+| batch-1.0
+| batch-2.0
+
+| batchManagement-1.0
+| batchManagement-2.0
+
+| beanValidation-2.0
+| beanValidation-3.0
+
+| cdi-2.0
+| cdi-3.0
+
+| concurrent-1.0
+| concurrent-2.0
+
+| javaee-8.0
+| jakartaee-9.0
+
+| javaeeClient-8.0
+| jakartaeeClient-9.0
+
+| jsonb-1.0
+| jsonb-2.0
+
+| jsonbContainer-1.0
+| jsonbContainer-2.0
+
+| jsonp-1.1
+| jsonp-2.0
+
+| jsonpContainer-1.1
+| jsonpContainer-2.0
+
+| managedBeans-1.0
+| managedBeans-2.0
+
+| mdb-3.2
+| mdb-4.0
+
+| servlet-4.0
+| servlet-5.0
+
+| webProfile-8.0
+| webProfile-9.0
+
+| websocket-1.1
+| websocket-2.0
+|===
+
+In other cases, in addition to updating the feature version, the feature short name was updated.
```

Comment:
Again I would combine these two sentences into the same paragraph.  The content of the second sentence is incorrect.  Here is what I suggest:

`In other cases, in addition to updating the feature version, the feature short name was updated to use the new Jakarta short names of the feature.  If you use the the previous feature name with a new version, the server will not start the feature, but will inform you of the new feature names that are outlined in the table below.`

Reply:
Your version of the update does not say it will give you the correct feature name.  It just says it will tell you an updated feature name must be used.  But we actually tell you what that feature name is.  I suggest updating to something like:

`If the incorrect Jakarta EE feature short name is used with a new Jakarta EE 9 version, for example, changing "ejb-3.2" to "ejb-4.0", the server will not start and the server logs will provide information to say which new feature name must be used.`

Reply:
Updated.  The update ended up being a substantial rewrite of the paragraph, so see the new text.

Reply:
I would replace `, if` with `and`

Reply:
Rewrote the paragraphs for clarity.